src/js/controllers/imageCtrl-test.js

```diff
@@ -0,0 +1,142 @@
+
+                it('parent object should contain a photos object', function () {
+                    expect(angular.isObject(vm.images.photos)).to.equal(true);
+                });
+
+                it('photos object should include an photo object', function(){
+                    expect(angular.isObject(vm.images.photos.photo)).to.equal(true);
+                });
+
+            });
+
+            describe('individual photo object', function(){
+                var photo;
+
+                beforeEach(function(){
+                    photo = vm.images.photos.photo;
+                });
+
+                it('check for neccesary attibutes', function(){
+                    angular.forEach(photo, function(value){
+                        expect(value.farm).to.exist();
+                        expect(value.server).to.exist();
+                        expect(value.id).to.exist();
+                        expect(value.owner).to.exist();
+                        expect(value.secret).to.exist();
+                    });
+                });
+
+                it('check for neccesary attibutes to not be undefined ', function(){
+                    angular.forEach(photo, function(value){
+                        expect(value.farm).not.to.be.undefined();
+                        expect(value.server).not.to.be.undefined();
+                        expect(value.id).not.to.be.undefined();
+                        expect(value.owner).not.to.be.undefined();
+                        expect(value.secret).not.to.be.undefined();
+                    });
+                });
+
+                it('check for neccesary attibutes to not be null ', function(){
+                    angular.forEach(photo, function(value){
+                        expect(value.farm).not.to.be.null();
+                        expect(value.server).not.to.be.null();
+                        expect(value.id).not.to.be.null();
+                        expect(value.owner).not.to.be.null();
+                        expect(value.secret).not.to.be.null();
+                    });
+                });
+
+            });
+        });
+
+        describe('Error from to api', function() {
+
+            beforeEach(module(function($provide){
+                $provide.service('images', function($q, $log){
+                    this.loadPics = sinon.stub().returns($q.reject($log));
+                });
+
+            }));
+
+            beforeEach(inject(function (_$rootScope_, _$controller_, _images_, _$q_, _$log_) {
+                $rootScope = _$rootScope_.$new();
+                $controller = _$controller_;
+                images = _images_;
+                $q = _$q_;
+                $log = _$log_;
+                vm = $controller('imagesCtrl');
+
+                $rootScope.$apply();
+            }));
+
+            it('Should log error message ', function(){
+                var errorMsg = 'Error loading images ';
+                expect($log.error.logs[0]).to.include(errorMsg);
```

Comment:
Not a bad way to test whether the $log has been called. Typically, I use a spy here and check what against the string value of the message.